### PR TITLE
Accounts Revamp Fixes: "Overall" Wallet Improvements

### PR DESCRIPTION
**Pull request by rauljordan**
> Bug fix

**What does this PR do? Why is it needed?**

- [x] Changes the wallet dir default path to not be a hidden directory
- [x] add `WalletPasswordsDirFlag` to `edit-config`
- [x]  It would be nice if the wallet's password path could be saved as part of the wallet's metadata. Typing this path over and over is super annoying.
- [x] Whenever I do not provide at least one flag out of the 4 gRPC configuration flags during create or edit-config, I am prompted for all 4 configuration values and I have to pass them again
- [x] Add a --wallet-password-file and a --account-password-file flag
- [x] I don't understand how to use the --password-flle flag based on the help available in the console.
- [x] I think it should say exported accounts instead of exported wallet. It would also be nice if the default path was dynamically adjusted to reflect the path from the --wallet-dir flag.
- [x]  Why does the export command contain a --passwords-dir flag when the passwords directory is not exported? Maybe it should be? (Nishant: I dont think we should be exporting passwords in batch, imo just exporting the wallets should be ok. Also
- [x] There is a small difference in formatting between list for a non-HD wallet and an HD wallet. #6725
- [x] Account slist with no wallet shouldn't panic
- [x] Panics in https://gist.github.com/prestonvanloon/48b134de6c000acaf01671249ac3aa94
- [x] Import creates a new wallet if none exists

**Which issues(s) does this PR fix?**

Part of #6715 

**Other notes for review**

### CODE REVIEW

File: validator/accounts/v2/accounts_import.go [hunk 1]
```diff
@@ -23,24 +23,24 @@ import (
...
 		}
-		if keymanagerKind != v2keymanager.Direct {
-			return fmt.Errorf(
-				"importing non-HD accounts into a non-direct wallet is not allowed, given wallet path contains a %s wallet",
-				keymanagerKind.String(),
-			)
+		if err = createDirectKeymanagerWallet(cliCtx, w); err != nil {
+			return errors.Wrap(err, "could not initialize wallet")
 		}
+		log.WithField("wallet-path", w.walletDir).Infof(
```
Comment (0xKiwi) [inline validator/accounts/v2/accounts_import.go:41]:
```suggestion
		log.WithField("wallet-path", w.walletDir).Info(
```

File: validator/accounts/v2/wallet.go [hunk 2]
```diff
@@ -60,6 +49,7 @@ var (
 // and providing secure access to eth2 secrets depending on an
 // associated keymanager (either direct, derived, or remote signing enabled).
 type Wallet struct {
+	walletDir      string
 	accountsPath   string
 	passwordsDir   string
```
Comment (0xKiwi) [inline validator/accounts/v2/wallet.go:54]: Whats the difference between walletDir and accountsPath?

File: validator/accounts/v2/wallet_create.go [hunk 3]
```diff
@@ -15,53 +15,55 @@ import (
 // CreateWallet from user input with a desired keymanager. If a
 // wallet already exists in the path, it suggests the user alternatives
 // such as how to edit their existing wallet configuration.
-func CreateWallet(cliCtx *cli.Context) error {
+func CreateWallet(cliCtx *cli.Context) (*Wallet, error) {
```
Comment (0xKiwi) [inline validator/accounts/v2/wallet_create.go:18]: Where is this returned wallet used?

Comment (rauljordan) [reply validator/accounts/v2/wallet_create.go:18]: It's used as part of account create or import

File: validator/accounts/v2/wallet.go (hunk 2 above)
Comment (rauljordan) [reply validator/accounts/v2/wallet.go:54]: Yeah, the wallet path is the path of the prysm-wallet-v2, and the accounts path is deeper within that directory

File: validator/accounts/v2/accounts_create.go [hunk 4]
```diff
@@ -6,12 +6,13 @@ import (
 
 	"github.com/manifoldco/promptui"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+
```
Comment (0xKiwi) [inline validator/accounts/v2/accounts_create.go:11]:
```suggestion
```

File: validator/flags/flags.go [hunk 5]
```diff
@@ -12,6 +12,17 @@ import (
...
 
+const (
+	// WalletDefaultDirName for accounts-v2.
+	WalletDefaultDirName = "prysm-wallet-v2"
+	// PasswordsDefaultDirName where account passwords are stored.
+	PasswordsDefaultDirName = "prysm-wallet-v2-passwords"
+	// KeymanagerConfigFileName for the keymanager used by the wallet: direct, derived, or remote.
+	KeymanagerConfigFileName = "keymanageropts.json"
+	// DirectoryPermissions for directories created under the wallet path.
+	DirectoryPermissions = os.ModePerm
```
Comment (0xKiwi) [inline validator/flags/flags.go:23]: Do you mind moving this back to wallet.go?

File: validator/accounts/v2/wallet.go [hunk 6]
```diff
@@ -474,6 +488,27 @@ func readKeymanagerKindFromWalletPath(walletPath string) (v2keymanager.Kind, err
 	return v2keymanager.ParseKind(list[0])
 }
 
+func createOrOpenWallet(cliCtx *cli.Context, creationFunc func(cliCtx *cli.Context) (*Wallet, error)) (*Wallet, error) {
```
Comment (prestonvanloon) [inline validator/accounts/v2/wallet.go:491]: Please test

File: validator/accounts/v2/wallet_edit.go [hunk 7]
```diff
@@ -21,7 +23,29 @@ func EditWalletConfiguration(cliCtx *cli.Context) error {
...
+		enc, err := wallet.ReadKeymanagerConfigFromDisk(ctx)
+		if err != nil {
+			return errors.Wrap(err, "could not read config")
+		}
+		cfg, err := direct.UnmarshalConfigFile(enc)
+		if err != nil {
+			return errors.Wrap(err, "could not unmarshal config")
+		}
+		log.Infof("Current configuration")
+		fmt.Printf("%s\n", cfg)
```
Comment (prestonvanloon) [inline validator/accounts/v2/wallet_edit.go:35]: Hmmm! Seems out of place

File: validator/accounts/v2/wallet_edit.go [hunk 8]
```diff
@@ -21,7 +23,29 @@ func EditWalletConfiguration(cliCtx *cli.Context) error {
...
-		return errors.New("no configuration options available to edit for direct keymanager")
+		enc, err := wallet.ReadKeymanagerConfigFromDisk(ctx)
+		if err != nil {
+			return errors.Wrap(err, "could not read config")
+		}
+		cfg, err := direct.UnmarshalConfigFile(enc)
+		if err != nil {
+			return errors.Wrap(err, "could not unmarshal config")
+		}
+		log.Infof("Current configuration")
```
Comment (prestonvanloon) [inline validator/accounts/v2/wallet_edit.go:34]: info instead of Infof

File: validator/accounts/v2/wallet_edit.go (hunk 7 above)
Comment (rauljordan) [reply validator/accounts/v2/wallet_edit.go:35]: This prints out the current configuration to stdout. Left a comment to clarify

**Pull request merged**